docs/content/sips/013-cloud-plugin.md

```diff
@@ -0,0 +1,161 @@
+simple on-ramp experience so users can readily deploy their Spin applications to
+a production-grade system.
+
+Spin’s existing plug-in system allows the Spin community to add and extend the
+functionality of Spin without introducing changes to the Spin command-line
+interface. Plug-ins are described in more detail in [SIP 006 - Spin
+Plugins](./006-spin-plugins.md). This allows us to ship `spin login` and `spin
+deploy` as separate, discrete functionality while ensuring users can still
+access its functionality through familiar tooling.
+
+## Specification
+
+The proposal herein suggests that we re-release the core logic of `spin login`
+and `spin deploy` under a separate `spin cloud` plugin. Spin will alias the
+existing `spin login` and `spin deploy` commands to their respective `spin
+cloud` counterparts to retain backwards compatibility.
+
+When a user executes `spin login` or `spin deploy` and the cloud plugin is not
+installed on their system, Spin will inform the user that the `spin cloud`
+plugin has not been installed, then install the plugin.
+
+```console
+$ spin deploy
+The `cloud` plugin is required. Installing now.
+Plugin 'cloud' was installed successfully!
+Uploading cloud_start version 0.1.0+XXXXXXXX...
+Deploying...
+Waiting for application to become ready... ready
+Available Routes:
+  cloud-start: https://cloud-start-xxxxxxxx.fermyon.app/ (wildcard)
+```
+
+## Future design considerations
+
+### `spin cloud config`
+
+An early prototype of the `spin cloud` plugin proposed several changes to the
+`spin cloud deploy` command, including a new `spin cloud config` command. The
+proposed command would configure the "current" Spin app for deployment with an
+interactive prompt. This command would be optional, and calling `spin cloud
+deploy` on an application that was not yet configured would invoke `spin cloud
+configure` in the background. The experience looked something like this:
+
+```console
+$ spin cloud deploy
+The current Spin app hasn't been set up for deployment yet.
+Pick a new or existing deployment config:
+
+> Fermyon Cloud - New Deployment
+  Fermyon Cloud - 'spicy-meatballs'
+
+Creating new deployment...
+New deployment created: 'mighty-hamster'
+
+[...SNIP normal deploy flow...]
+```
+
+While the move to a separate `spin cloud` plugin does not reject this idea
+outright, the goal of this SIP is to make the least invasive change to the
+existing `spin login` and `spin deploy` experience. Future iterations to the
+`spin login` and `spin deploy` experience can be addressed in future updates to
+the plugin.
+
+In fact, the movement to a separate `spin cloud` plugin grants us the
+flexibility to make changes to the core `spin deploy` experience without forcing
+us to wait until Spin 2.0. If anything, this proposal enables us to make these
+changes to the `spin cloud` plugin without waiting for a new release of Spin.
+
+### A generic `spin cloud` plugin supporting multiple cloud providers
+
```

Comment:
@bacongobbler I believe this is not a future we are sure about. Can we remove this section on the generic plugin?